src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
+    navContent: React.PropTypes.shape({
+      duration: React.PropTypes.number
+      label: React.PropTypes.string.isRequired,
+      onNextClick: React.PropTypes.func.isRequired,
+      onPreviousClick: React.PropTypes.func.isRequired
+    }),
+    onClose: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      duration: 500,
+      isOpen: false,
+      easing: [0.28, 0.14, 0.34, 1.04]
+    };
+  },
+
+  componentDidMount () {
+    this._renderTransition(this.props.isOpen);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.isOpen !== this.props.isOpen) {
+      this._renderTransition(newProps.isOpen);
+    }
+  },
+
+  _renderNavContent () {
```

Comment:
I'd either eliminate this separate render method and do everything in the JSX, or pull the conditional used to render this content out of the JSX and into this method.